File: Arrays/1. Find pair with given sum in the array/python/Sorting.py
# arr = input("arr: ")
# sum = input("sum: ")
arr = '8 7 2 5 3 1'
sum = '10'
sum = int(sum)
arr = list(map(int,arr.split(" ")))

print('arr: ', arr, type(arr))
print('sum: ', sum, type(sum))

def findPair(arr, sum):
    arr.sort()
    # arr = '1 2 3 5 7 8'

    (low, high) = (0,len(arr)-1)
    while (low < high):
        if (arr[low] + arr[high] == sum):
            print(arr[low], arr[high])
            low = low + 1

        elif (arr[low] + arr[high] < sum):
            low = low + 1

        elif (arr[low] + arr[high] > sum):
            high = high - 1
    
findPair(arr,sum)

File: Arrays/1. Find pair with given sum in the array/c++/Sorting.cpp
#include <iostream>
#include <algorithm>
#include <utility>
#include <vector>
#include <sstream>

using namespace std;

void printPair(pair<int, int> pair)
{
    cout << "(" << pair.first << ", " << pair.second << ")" << endl;
}

void findPair(vector<int> arr, int sum)
{
    sort(arr.begin(), arr.end());

    int low = 0;
    int high = arr.size() - 1;

    while (low < high)
    {
        if (arr[low] + arr[high] == sum)
        {
            printPair(pair<int, int>(arr[low], arr[high]));
            low = low + 1;
        }
        else if (arr[low] + arr[high] < sum)
        {
            low = low + 1;
        }
        else if (arr[low] + arr[high] > sum)
        {
            high = high - 1;
        }
    }
}

int main()
{
    string input;
    getline(cin, input);
    stringstream ss(input);
    string temp;

    vector<int> arr;

    while (getline(ss, temp, ' '))
    {
        arr.push_back(stoi(temp));
    }

    int sum;
    scanf("%d", &sum);

    // input example = 8 7 2 5 3 1 ; sum = 10

    findPair(arr, sum);

    return 0;
}